src/components/UserProfile/TeamProfile.tsx

```diff
@@ -0,0 +1,155 @@
+"use client";
+
+import { generateClient } from "aws-amplify/data";
+import { useState } from "react";
+
+import { type Schema } from "@/amplify/data/resource";
+import ProfileLinks from "@/components/UserProfile/ProfileLinks";
+import TeamForm from "@/components/UserProfile/TeamForm";
+import { useUser } from "@/components/contexts/UserContext";
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+
+const BUTTON_STYLES =
+  " rounded-full border-4 border-white bg-[#FF6B54] px-10  md:px-12 py-2 my-2 text-white";
+
+const TEAM_INSTRUCTION_STYLES =
+  "bg-pink bg-white/30 mx-10 my-10 rounded-3xl  border-4 border-white bg-[#FFFFFF] px-10 py-20 md:px-20 md:py-16";
+
+const client = generateClient<Schema>();
+
+export interface TeamFormProp {
+  data: Schema["Team"]["type"];
+  setHasTeam: React.Dispatch<React.SetStateAction<boolean>>;
+  teamMutation: any;
+}
+
+const TeamProfile = () => {
+  const queryClient = useQueryClient();
+  const [hasTeam, setHasTeam] = useState<boolean>(false);
+
+  const userId = useUser().currentUser.userSub as string;
+
+  const { data, isFetching } = useQuery({
+    initialData: {} as Schema["Team"]["type"],
+    initialDataUpdatedAt: 0,
+    queryKey: ["Team", userId],
+    queryFn: async () => {
+      const userResponse = await client.models.User.get({
+        id: userId,
+      });
+
+      const userTeamId = userResponse.data?.teamId as string;
+
+      if (!userTeamId) {
+        return;
+      }
```

Comment:
This ties into the comment below but to fix the ESLint error on Line 33, we can make this line look like this: 
```suggestion
        return {} as Schema["Team"]["type"];
```